src/m365/teams/commands/message/message-export.ts

```diff
@@ -0,0 +1,218 @@
+
+  public get name(): string {
+    return commands.MESSAGE_EXPORT;
+  }
+
+  public get description(): string {
+    return 'Export Microsoft Teams chat messages for a given user, or a team.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        teamId: typeof args.options.teamId !== 'undefined',
+        fromDateTime: typeof args.options.fromDateTime !== 'undefined',
+        toDateTime: typeof args.options.toDateTime !== 'undefined',
+        withAttachments: !!args.options.withAttachments,
+        folderPath: typeof args.options.folderPath !== 'undefined'
+      });
+    });
+  }
+
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '--teamId [teamId]'
+      },
+      {
+        option: '--fromDateTime [fromDateTime]'
+      },
+      {
+        option: '--toDateTime [toDateTime]'
+      },
+      {
+        option: '--withAttachments'
+      },
+      {
+        option: '--folderPath [folderPath]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid userPrincipalName`;
+        }
+
+        if (args.options.teamId && !validation.isValidGuid(args.options.teamId)) {
+          return `${args.options.teamId} is not a valid GUID`;
+        }
+
+        if (args.options.fromDateTime && !validation.isValidISODateTime(args.options.fromDateTime)) {
+          return `${args.options.fromDateTime} is not a valid ISO DateTime`;
+        }
+
+        if (args.options.toDateTime && !validation.isValidISODateTime(args.options.toDateTime)) {
+          return `${args.options.toDateTime} is not a valid ISO DateTime`;
+        }
+
+        if (args.options.folderPath && !fs.existsSync(args.options.folderPath)) {
+          return `Path ${args.options.folderPath} does not exist.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['userId', 'userName', 'teamId']
+      },
+      {
+        options: ['folderPath'],
+        runsWhen: (args: CommandArgs) => {
+          return args.options.withAttachments !== undefined && args.options.withAttachments;
```

Comment:
You're checking twice the same here? You could also just use `=== true` as well

Reply:
Correct